### PR TITLE
Add GLUE submission formatter script and task functionality

### LINKED ISSUE #109: Add support for exporting GLUE task test predictions in leaderboard submission format
This issue tracks progress on adding support for exporting GLUE task test predictions in leaderboard submission format (see "[How do I use GLUE](https://gluebenchmark.com/faq)" for details of submission format).

### PR DESCRIPTION
**Pull request by jeswan**
This PR adds a GLUE submission formatter script and closes #109. This script uses the output (test_preds.p) of jiant's `runscript.py --write_test_preds` and converts it to the format expected by https://gluebenchmark.com/.

**How were these changes tested?**
All GLUE tasks were run E2E (download, tokenized, fine-tuned) using ~0.01 epochs of training. Then, the glue_submission_formatter.py script was used to generate the required .tsv's for submission. All .tsv output files were manually inspected and a complete submission will be tested before merging this PR (QQP requires a couple of hours to generate test predictions).

### CODE REVIEW

File: jiant/tasks/lib/stsb.py [hunk 1]
```diff
@@ -114,3 +114,13 @@ def _create_examples(cls, lines, set_type):
...
+
+    @classmethod
+    def get_glue_preds(cls, pred_dict):
+        """Returns a tuple of (index, prediction) as expected by GLUE."""
+        indexes = []
+        predictions = []
+        for pred, guid in zip(list(pred_dict["preds"]), list(pred_dict["guids"])):
+            indexes.append(int(guid.split("-")[1]))
+            predictions.append(str(round(pred, 3)))
+        return (indexes, predictions)
```
Comment (jeswan) [inline jiant/tasks/lib/stsb.py:126]: override `get_glue_preds()` since output is float.

**Comment by zphang** [comment]
Quick questions:

1. Why does MnliMismatched need to have its own implementation?
2. Is it necessary for the GlueTask to be a superclass? It feels like a mixin would be better (e.g. RTE will fall under both GLUE and SuperGLUE)
3. Has `glue_submission_formatter.py` not yet been added?

**Comment by jeswan** [comment]
@zphang 

1. It doesn't necessarily...it can be removed. Was trying to simplify "mapping" between tasks.
2. I'll take a look at mixins (I don't have much experience). Likely a better option based on your observation.
3. Just added

**Comment by jeswan** [comment]
Outputs from script were successfully uploaded to https://gluebenchmark.com/

File: jiant/scripts/postproc/glue_submission_formatter.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    parser.add_argument(
+        "--input_base_path",
+        required=True,
+        help="base input path of GLUE tasks (contains task folders)",
+    )
+    parser.add_argument("--output_path", required=True, help="output path for formatted files")
+
+    args = parser.parse_args()
+
+    # for task_name, input_filepath in raw_pred_input_filepaths.items():
```
Comment (zphang) [inline jiant/scripts/postproc/glue_submission_formatter.py:43]: Delete?

File: jiant/scripts/postproc/glue_submission_formatter.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+"""Translate raw prediction files for GLUE tasks into format expected by GLUE leaderboard.
+This script translates raw prediction files for GLUE tasks into the tsv files required
+by the GLUE leaderboard. See https://super.gluebenchmark.com/faq for leaderboard info.
```
Comment (zphang) [inline jiant/scripts/postproc/glue_submission_formatter.py:3]: GLUE not SuperGLUE

File: jiant/scripts/postproc/glue_submission_formatter.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        required=True,
+        help="base input path of GLUE tasks (contains task folders)",
+    )
+    parser.add_argument("--output_path", required=True, help="output path for formatted files")
+
+    args = parser.parse_args()
+
+    # for task_name, input_filepath in raw_pred_input_filepaths.items():
+    for task_name in GLUE_TASKS:
+        input_filepath = os.path.join(args.input_base_path, task_name, "test_preds.p")
```
Comment (zphang) [inline jiant/scripts/postproc/glue_submission_formatter.py:45]: Add a README for documenting how to run this? (Can be a later PR)

File: jiant/scripts/postproc/glue_submission_formatter.py (hunk 3 above)
Comment (zphang) [reply jiant/scripts/postproc/glue_submission_formatter.py:3]: (Referring to the URL)

File: jiant/scripts/postproc/glue_submission_formatter.py (hunk 4 above)
Comment (jeswan) [reply jiant/scripts/postproc/glue_submission_formatter.py:44]: Great idea! I'll add that as a subsequent PR shortly.

Comment (jeswan) [reply jiant/scripts/postproc/glue_submission_formatter.py:44]: Great point...will add this is a second PR.

**Comment by jeswan** [comment]
@zphang added an option for evaluating on a subset of GLUE tasks

**Pull request merged**